### PR TITLE
[checkify] fix corner case where debug_info must be updated within custom_jvp_call

### PR DESCRIPTION
**Pull request by jakevdp**
This came up while testing the hijax execution path for searchsorted.

### CODE REVIEW

File: tests/checkify_test.py [hunk 1]
```diff
@@ -956,6 +956,26 @@ def fn(x: jax.Array):
...
+      return f(x), dx * 2.0
+
+    def body(x):
+      checkify.check(x > 0, "x must be positive")
+      return f(x)
+
+    checked_f = jax.jit(checkify.checkify(body))
+    err, res = checked_f(2.0)
+    self.assertIsNone(err.get())
+    self.assertAllClose(res, 4.0)
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply tests/checkify_test.py:977]: The proposed test passes without changes to the vjp execution path.

File: jax/_src/checkify.py [hunk 2]
```diff
@@ -1038,10 +1038,13 @@ def custom_jvp_call_rule(in_err: Error,
   # JVP rule, one must instead use checkify-of-jvp. Thus this implementation
   # just forwards the input error and code (and trivial tangents) to the output.
   err_vals, err_tree = jtu.tree_flatten(in_err)
+  dbg = call_jaxpr.debug_info
+  if dbg.arg_names is not None:
+    dbg = dbg._replace(arg_names=("",) * err_tree.num_leaves + dbg.arg_names)
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply jax/_src/checkify.py:1043]: I checked, and vjp doesn't seem to have the same issue.
